flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -351,9 +351,11 @@ object ScalarOperatorGens {
     checkImplicitConversionValidity(left, right)
     val canEqual = isInteroperable(left.resultType, right.resultType)
     if (isCharacterString(left.resultType) && isCharacterString(right.resultType)) {
-      generateOperatorIfNotNull(ctx, new BooleanType(), left, right) {
-        (leftTerm, rightTerm) => s"$leftTerm.equals($rightTerm)"
-      }
+      generateOperatorIfNotNull(
+        ctx,
+        new BooleanType(left.resultType.isNullable || right.resultType.isNullable),
```

Comment:
I think it would be more accurate to pass in resultType. Inferring type twice is hard to guarantee the type inferring is the same.

Reply:
Could you fix all the expression generation by passing in `resultType`? You can see that most expressions are using the given `resultType` in `org.apache.flink.table.planner.codegen.ExprCodeGenerator#generateCallExpression`.

Reply:
I have fix all the expressions. Hope I haven't miss any expression.

Reply:
I have fix all the expressions. Hope I haven't miss any expression.